independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanConfigurator.java

```diff
@@ -43,6 +43,31 @@ public void done() {
             if (implClass == null) {
                 throw new IllegalStateException("Unable to find the bean class in the index: " + implClazz);
             }
+
+            ScopeInfo scope = this.scope;
+            if (scope == null) {
+                scope = Beans.initStereotypeScope(stereotypes, implClass, beanDeployment);
+            }
+            if (scope == null) {
+                scope = BuiltinScope.DEPENDENT.getInfo();
+            }
+
+            String name = this.name;
+            if (name == null) {
+                name = Beans.initStereotypeName(stereotypes, implClass);
+            }
+
+            Boolean alternative = this.alternative;
+            if (alternative == null) {
+                alternative = Beans.initStereotypeAlternative(stereotypes);
+            }
+
+            Integer priority = this.priority;
+            if (priority == null) {
+                // TODO use `Beans.initAlternativePriority` instead? IMHO no for synthetic beans
```

Comment:
I think that `Beans.initAlternativePriority()` is purely about setting the priority to `Integer.MAX_VALUE` for alternatives selected via the `quarkus.arc.selected-alternatives` config property. Which probably does not make sense for synthetic beans...

Reply:
Good point, I didn't look deep enough when writing this. Will remove the comment.